File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Calendars/removecalendarexception.rb
module Asposetasksjava
  module RemoveCalendarException
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Remove an exception
        cal = project.getCalendars().toList().get(0)
        if cal.getExceptions().getCount() > 1
            exc = cal.getExceptions().toList().get(0)
            cal.getExceptions().remove(exc)
            puts "Removed calendar exception."
        end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithCalendarExceptions/RemoveCalendarException.php
<?php
namespace Aspose\Tasks\WorkingWithCalendarExceptions;

use com\aspose\tasks\Project as Project;

class RemoveCalendarException{
    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Remove an exception
        $cal = $project->getCalendars()->toList()->get(0);
        if((int)(string)($cal->getExceptions()->getCount()) > 1) {
            $exc = $cal -> getExceptions() -> toList() -> get(0);
            $cal -> getExceptions() -> remove($exc);
            print "Removed calendar exception.".PHP_EOL;
        }

    }

}
